runtime/src/main/scala/akka/grpc/scaladsl/Grpc.scala

```diff
@@ -0,0 +1,41 @@
+package akka.grpc.scaladsl
+
+import scala.concurrent.Future
+import scala.jdk.CollectionConverters.asScalaSetConverter
+
+import akka.Done
+import akka.actor.{ CoordinatedShutdown, ExtendedActorSystem, Extension, ExtensionId, ExtensionIdProvider }
+import akka.annotation.InternalApi
+import akka.grpc.internal.ClientState
+
+import java.util.concurrent.ConcurrentHashMap
+
+/** INTERNAL API */
+@InternalApi
+class GrpcImpl(system: ExtendedActorSystem) extends Extension {
+  private val clients = new ConcurrentHashMap[ClientState, Unit]
+
+  CoordinatedShutdown(system).addTask("before-actor-system-terminate", "close-grpc-clients") { () =>
+    implicit val ec = system.dispatcher
+    Future.sequence(clients.keySet().asScala.map(_.close())).map(_ => Done)
```

Comment:
I think with sequence if one future op fails it won't execute the next so maybe wrap the close with a try just for good measure?